From 055b16344d4cadad0f248978e96290cf6ff70727 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Wed, 19 Jun 2024 15:35:48 +0200
Subject: [PATCH 0954/1695] dts: stm32: declare RISAB instances as firewall
 controllers on stm32mp25

Declare RISABs as access controllers so that it's possible to check
firewall access rights on address ranges, and reconfigure dynamically
RISAB regions.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git fbfd3c59e092f27aa8e7fabd2e5596983c930141 dts: stm32: declare RISAB instances as firewall controllers on stm32mp25]
Change-Id: Ib6a7ed141315ec4fa6ea2364fa44035640982184
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/389863
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/arch/arm/dts/stm32mp251.dtsi | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp251.dtsi b/core/arch/arm/dts/stm32mp251.dtsi
index 0afbe3af3..a7921e179 100644
--- a/core/arch/arm/dts/stm32mp251.dtsi
+++ b/core/arch/arm/dts/stm32mp251.dtsi
@@ -695,6 +695,7 @@
 			reg = <0x420f0000 0x1000>;
 			clocks = <&rcc CK_ICN_LS_MCU>;
 			st,mem-map = <0xa000000 0x20000>;
+			#access-controller-cells = <1>;
 		};
 
 		risab2: risab@42100000 {
@@ -702,6 +703,7 @@
 			reg = <0x42100000 0x1000>;
 			clocks = <&rcc CK_ICN_LS_MCU>;
 			st,mem-map = <0xa020000 0x20000>;
+			#access-controller-cells = <1>;
 		};
 
 		risab3: risab@42110000 {
@@ -709,6 +711,7 @@
 			reg = <0x42110000 0x1000>;
 			clocks = <&rcc CK_ICN_LS_MCU>;
 			st,mem-map = <0xa040000 0x20000>;
+			#access-controller-cells = <1>;
 		};
 
 		risab4: risab@42120000 {
@@ -716,6 +719,7 @@
 			reg = <0x42120000 0x1000>;
 			clocks = <&rcc CK_ICN_LS_MCU>;
 			st,mem-map = <0xa060000 0x20000>;
+			#access-controller-cells = <1>;
 		};
 
 		risab5: risab@42130000 {
@@ -723,6 +727,7 @@
 			reg = <0x42130000 0x1000>;
 			clocks = <&rcc CK_ICN_LS_MCU>;
 			st,mem-map = <0xa080000 0x20000>;
+			#access-controller-cells = <1>;
 		};
 
 		risab6: risab@42140000 {
@@ -730,6 +735,7 @@
 			reg = <0x42140000 0x1000>;
 			clocks = <&rcc CK_ICN_LS_MCU>;
 			st,mem-map = <0xa0a0000 0x20000>;
+			#access-controller-cells = <1>;
 			status = "disabled";
 		};
 
-- 
2.47.2

